sys/include/net/sock/ip.h

```diff
@@ -12,7 +12,244 @@
+ * don't. A client that does not require a reply is very simple to implement in
+ * one line:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * res = sock_ip_send(NULL, data, data_len, PROTNUM, &remote);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * With `data` being the data send, `data_len` the length of `data`, `PROTNUM`
+ * the next header number for the send packet and `remote` the remote end point
+ * the packet is to be sent.
+ *
+ * To see some other utilization we look at a more complex example in form of
+ * the counter of the echo server above:
+ *
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #include <stdio.h>
+ *
+ * #include "net/af.h"
+ * #include "net/protnum.h"
+ * #include "net/ipv6/addr.h"
+ * #include "net/sock/ip.h"
+ * #include "xtimer.h"
+ *
+ * uint8_t buf[7];
+ *
+ * int main(void)
+ * {
+ *     sock_ip_ep_t local = SOCK_IPV6_EP_ANY;
+ *     sock_ip_t sock;
+ *
+ *     if (sock_ip_create(&sock, &local, NULL, PROTNUM_IPV6_NONXT, 0) < 0) {
+ *         puts("Error creating raw IP sock");
+ *         return 1;
+ *     }
+ *
+ *     while (1) {
+ *         sock_ip_ep_t remote = { .family = AF_INET6 };
+ *         ssize_t res;
+ *
+ *         ipv6_addr_set_all_nodes_multicast((ipv6_addr_t *)&remote.addr.ipv6,
+ *                                           IPV6_ADDR_MCAST_SCP_LINK_LOCAL);
+ *
+ *         if (sock_ip_send(&sock, "Hello!", sizeof("Hello!"), 0, &remote) < 0) {
+ *             puts("Error sending message");
+ *             sock_ip_close(&sock);
+ *             return 1;
+ *         }
+ *         if ((res = sock_ip_recv(&sock, buf, sizeof(buf), 1 * SEC_IN_USEC,
+ *                                 NULL)) < 0) {
+ *             if (res == -ETIMEDOUT) {
+ *                 puts("Timed out");
+ *             }
+ *             else {
+ *                 puts("Error receiving message");
+ *             }
+ *         }
+ *         else {
+ *             printf("Received message: \"");
+ *             for (int i = 0; i < res; i++) {
+ *                 printf("%c", buf[i]);
+ *             }
+ *             printf("\"\n");
+ *         }
+ *         xtimer_sleep(1);
+ *     }
+ *
+ *     return 0;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * Again: Don't forget to also @ref including-modules "include" the IPv6 module
+ * of your networking implementation (e.g. `gnrc_ipv6_default` for
+ * @ref net_gnrc GNRC) and at least one network device.
+ *
+ * We first create again a `sock` with a local end point bound to any IPv6
+ * address. Note that we also could set the remote here and not use it with
+ * @ref sock_ip_send().
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ *     sock_ip_ep_t local = SOCK_IPV6_EP_ANY;
+ *     sock_ip_t sock;
+ *
+ *     if (sock_ip_create(&sock, &local, NULL, PROTNUM_IPV6_NONXT, 0) < 0) {
+ *         puts("Error creating raw IP sock");
+ *         return 1;
+ *     }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * We then create a remote end point for the link-local all nodes multicast
```

Comment:
if I understand well (I hope I do ;) ), I would rephrase like this : 
create a remote end point with the link-local multicast address of all nodes and send